doc/nrf/ug_bootloader_adding.rst

```diff
@@ -0,0 +1,436 @@
+This is useful when the private keys are stored, managed, or otherwise processed through a *hardware security module* (`HSM`_) or in-house tool.
+
+To use a custom signing command with this bootloader, set the following options in a ``prj.conf`` file or Kconfig fragment:
+
+.. code-block:: console
+
+   CONFIG_SECURE_BOOT=y
+   CONFIG_SB_SIGNING_CUSTOM=y
+   CONFIG_SB_SIGNING_PUBLIC_KEY="/path/to/pub.pem"
+   CONFIG_SB_SIGNING_COMMAND="my_command"
+
+You can also use the command line, as follows:
+
+.. code-block:: console
+
+   west build -b nrf52840dk_nrf5840 zephyr/samples/hello_world -- \
+   -DCONFIG_SECURE_BOOT=y \
+   -DCONFIG_SB_SIGNING_CUSTOM=y \
+   -DCONFIG_SB_SIGNING_PUBLIC_KEY=\"/path/to/pub.pem\" \
+   -DCONFIG_SB_SIGNING_COMMAND=\"my_command\"
+
+The public key string must be an absolute path to the location of the public key file.
+Environment variables and the ``~`` character on Unix systems will not be expanded when setting this value from a ``prj.conf`` file or Kconfig fragment, but will be expanded on the command line.
+
+See :option:`CONFIG_SB_SIGNING_COMMAND` for specifics about what a signing command must do to be usable.
+The command string can include its own arguments like a typical terminal command, with arguments sent to the command by the build system being appended to the end:
+
+.. parsed-literal::
+   :class: highlight
+
+   my_command *[options]* *<args ...>* *<build_system_args ..>*
+
+Arguments sent to the command by the build system in this way are noted in :option:`CONFIG_SB_SIGNING_COMMAND`.
+
+.. note::
+
+   Whitespace, hyphens, and other non-alphanumeric characters must be escaped appropriately when setting the string from the command line.
+   If the custom signing command uses its own options or arguments, it is recommended to define the string in a ``prj.conf`` file or Kconfig fragment to avoid tracking backslashes.
+   Like public key paths, environment variables will not be expanded when using them in a command string set from one of these files.
+
+.. _ug_bootloader_adding_immutable_mcuboot:
+
+MCUboot
+=======
+
+To build :ref:`mcuboot:index` with a Zephyr or |NCS| sample, enable the :option:`CONFIG_BOOTLOADER_MCUBOOT` in the application's :file:`prj.conf` file.
+You can also use the command line, as follows:
+
+.. code-block:: console
+
+   west build -b nrf52840dk_nrf52840 zephyr/samples/hello_world -- -DCONFIG_BOOTLOADER_MCUBOOT=y
+
+|how_to_configure|
+Like other child images, you can assign :ref:`image-specific configurations <ug_multi_image_variables>` at build time to further customize the bootloader's functionality.
+
+.. _ug_bootloader_adding_immutable_mcuboot_keys:
+
+Adding a custom signature key file
+----------------------------------
+
+.. FIXME: Sphinx does not find :option:`CONFIG_BOOT_SIGNATURE_KEY_FILE`; how to link against non-NCS Kconfig options?
+
+To pass the signature key file into the MCUboot image, set its :option:`CONFIG_BOOT_SIGNATURE_KEY_FILE` to the generated file.
```

Comment:
`to the private key file` ? Generated file may sound a bit ambiguous

Reply:
fixed